aws/src/main/java/org/apache/iceberg/aws/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,201 @@
+          GetTemporaryGlueTableCredentialsRequest.builder()
+              .tableArn(tableArn)
+              .permissions(Permission.ALL)
+              .build();
+      GetTemporaryGlueTableCredentialsResponse response =
+          client.getTemporaryGlueTableCredentials(getTemporaryGlueTableCredentialsRequest);
+      return AwsSessionCredentials.create(response.accessKeyId(), response.secretAccessKey(), response.sessionToken());
+    }
+
+    private static AwsCredentialsProvider getAssumeRoleCredentialsProvider(Map<String, String> properties) {
+      String roleArn = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_ARN);
+      Preconditions.checkNotNull(roleArn, "Cannot initialize LakeFormationAwsClientFactory with empty roleArn");
+      String sessionName = String.format("iceberg-aws-%s", UUID.randomUUID());
+      String externalId = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_EXTERNAL_ID);
+      String tagString = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_TAGS_PREFIX);
+      Preconditions.checkNotNull(tagString, "Cannot initialize LakeFormationAwsClientFactory with empty tags");
+      Set<Tag> tag = toTags(properties);
+      int timeout = PropertyUtil.propertyAsInt(properties,
+          AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC, AwsProperties.CLIENT_ASSUME_ROLE_TIMEOUT_SEC_DEFAULT);
+      AssumeRoleRequest request = AssumeRoleRequest.builder()
+          .roleArn(roleArn)
+          .roleSessionName(sessionName)
+          .durationSeconds(timeout)
+          .externalId(externalId)
+          .tags(tag)
+          .build();
+      return StsAssumeRoleCredentialsProvider.builder()
+          .stsClient(StsClient.builder().httpClientBuilder(UrlConnectionHttpClient.builder()).build())
+          .refreshRequest(request)
+          .build();
+    }
+
+    private static Set<Tag> toTags(Map<String, String> properties) {
+      return PropertyUtil.propertiesWithPrefix(properties, AwsProperties.CLIENT_ASSUME_ROLE_TAGS_PREFIX)
+          .entrySet().stream()
+          .map(e -> Tag.builder().key(e.getKey()).value(e.getValue()).build())
+          .collect(Collectors.toSet());
+    }
+
+    private LakeFormationClient buildLakeFormationClient(AwsCredentialsProvider awsCredentialsProvider, String region) {
+      return LakeFormationClient.builder()
+          .credentialsProvider(awsCredentialsProvider)
+          .region(Region.of(region)).build();
+    }
+
+    private String getTableArn(Map<String, String> properties) {
+      String dbName = properties.get(AwsProperties.LAKE_FORMATION_DB_NAME);
+      String tableName = properties.get(AwsProperties.LAKE_FORMATION_TABLE_NAME);
+      String accountId = properties.get(AwsProperties.LAKE_FORMATION_ACCOUNT_ID);
+      String region = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_REGION);
+      String partitionName = PartitionMetadata.of(Region.of(region)).name();
+      return String.format("arn:%s:glue:%s:%s:table/%s/%s",
+          partitionName,
```

Comment:
`PartitionMetadata.of(Region.of(region)).name()` would return 
    /**
     * Returns the partition name for this partition, such as AWS Standard
     *
     * @return The name of this partition
     */
    String name();

you should use id() here instead 

    /**
     * Returns the identifier for this partition, such as aws.
     *
     * @return The identifier for this partition.
     */
    String id();

Reply:
nice catch, fixed. thanks!